base_layer/core/src/transaction.rs

```diff
@@ -503,9 +503,15 @@ impl Transaction {
         &self.body
     }
 
+    /// Returns the byte size or weight of a transaction
     pub fn get_weight(&self) -> Result<usize, MessageFormatError> {
         Ok(self.to_binary()?.len())
     }
+
+    /// Returns the total fee allocated to each byte of the transaction
+    pub fn get_fee_per_byte(&self) -> Result<f64, MessageFormatError> {
```

Comment:
this is called `fee_per_gram` in other parts of the codebase.

Reply:
This isn't 100% summetrical with the `calculate_fee` function, but that's ok, if we just name this `calculate_ave_fee_per_gram` or something,

Reply:
Function renamed.